### PR TITLE
workarounds to building latest LaunchMON and STAT on ppc64le

### PR DESCRIPTION
**Pull request by lee218llnl**
@becker33, this PR allows LaunchMON to build on ppcle64. It still doesn't run correctly, but it allows STAT to build on top of it and then we can test STAT using serial process attaching, rather than parallel job attaching via LaunchMON.

### CODE REVIEW

**Comment by lee218llnl** [comment]
@tgamblin or @mplegendre mind taking a look at this?

File: var/spack/repos/builtin/packages/launchmon/package.py [hunk 1]
```diff
@@ -25,25 +25,46 @@
...
-            "--with-rm=slurm")
+    def setup_environment(self, spack_env, run_env):
+        # These env var settings are temporary until the LaunchMON build
+        # system automatically detects the appropriate libraries
+        if self.spec.satisfies('arch=linux-redhat7-ppc64le'):
+            spack_env.set('LD_LIBRARY_PATH', '/opt/ibm/spectrum_mpi/lib')
+            spack_env.set('LDFLAGS',
+                          '-L/opt/ibm/spectrum_mpi/lib '
+                          ' -Wl,-rpath=/opt/ibm/spectrum_mpi/lib '
+                          '-lopen-rte -lopen-pal')
```
Comment (mplegendre) [inline var/spack/repos/builtin/packages/launchmon/package.py]: I don't think we should be merging this with the hardcoded spectrum MPI paths.  Will things work if you add a dependency on 'mpi', then pointed it at the spectrum MPI?  I have an uncommited spectrum MPI package in the Spack tcetmp install on rzmanta.

File: var/spack/repos/builtin/packages/launchmon/package.py [hunk 2]
```diff
@@ -25,25 +25,46 @@
...
-        make()
-        make("install")
+    def configure_args(self):
+        args = []
+        # A hack to turn on x perms for configure since it was added via patch
+        if self.spec.satisfies('@1.0.3b'):
+            import os
+            st = os.stat('./configure')
+            os.chmod('./configure', st.st_mode | 64)
+        if self.spec.satisfies('arch=linux-redhat7-ppc64le'):
```
Comment (mplegendre) [inline var/spack/repos/builtin/packages/launchmon/package.py]: If you add the above mpi dependency, then this should turn into a test for spectrum MPI, and then link to the launcher based on spectrum_mpi's prefix.

**Comment by adamjstewart** [comment]
@lee218llnl What's the status of this PR?

**Comment by lee218llnl** [comment]
@adamjstewart this is still WIP. I need to revisit this with #3042 in place.

**Comment by alalazo** [comment]
@lee218llnl I'm currently going through the list of open PRs. Is this something that you still want, now that #3042 is in?

**Comment by lee218llnl** [comment]
We can scrap this PR. @dongahn, we will probably want an official LaunchMON release supporting PPC64LE. I will put in a new Spack package PR with the new LaunchMON version when that is available.

**Comment by alalazo** [comment]
@lee218llnl Thanks for the update!

**Pull request closed without merging**